### DISCUSSION TITLE
Post to Notion

### DISCUSSION THREAD

**Discussion by dealingwith** [johnlindquist/kit#286] [Share]
<img width="418" alt="Screen Shot 2021-05-19 at 11 06 46 PM" src="https://user-images.githubusercontent.com/124208/118917657-f8c41e00-b8f6-11eb-9c5d-de7be826ec4c.png">

I thought the Notion API might be a fun thing to play with via Script Kit. I have a singular "tasks" database that I use to capture all manner of things, so I always want to insert pages into it in order to process them later. These scripts will require you to update the databaseID value in the code itself. I could see an expansion of this script to allow for the selection of a specific Notion database from a list.

The second script below takes a second input in order to insert content into the new page -- I was wondering if there is a way to allow multiline text input via Script Kit, which would make that script more robust.

You can get a Notion API token [here](https://developers.notion.com/) and find quick-start instructions [here](https://developers.notion.com/docs), as well as the API reference [here](https://developers.notion.com/reference/intro).

For a quick overview of the data model, [check out Chris Biscardi's video here](https://www.youtube.com/watch?v=apKz987OLcE&t=1s). For a more in-depth overview, check out [this Notion blog post](https://www.notion.so/blog/data-model-behind-notion).

[Install notion-add-task](https://scriptkit.com/api/new?name=notion-add-task&url=https://gist.githubusercontent.com/dealingwith/d8e701f2e0e5e64cb820103af96a08d2/raw/b2c11daf28ed7e4b40f2809bda428b527dd78b24/notion-add-task.js)

```js
let notionToken = await env('NOTION_USER_TOKEN')
let databaseID = "3859b567fda3464ea5a69d5ccb56274b"
let content = await arg('Enter Task')

let {data} = await post(
  'https://api.notion.com/v1/pages', {
  "parent": { "database_id": databaseID },
  "properties": {
    "Name": {
      "title": [
        {
          "text": {
            "content": content
          }
        }
      ]
    }
  }
},
{
  headers: {
    Authorization: `Bearer ${notionToken}`,
  }
})

let pageID = data.id.replace(/-/g, "");
let pageURL = `https://notion.so/${databaseID}?p=${pageID}`

copy(pageURL)
```

[Install notion-add-task-with-content](https://scriptkit.com/api/new?name=notion-add-task-with-content&url=https://gist.githubusercontent.com/dealingwith/99c4da33565dddb115b5c173e1e29f48/raw/0c5b31b56b0fcc39fa3f589f7e2dcd4312f5c53a/notion-add-task-with-content.js)

```js
let notionToken = await env('NOTION_USER_TOKEN')
let databaseID = "3859b567fda3464ea5a69d5ccb56274b"
let content = await arg('Enter Task')

let {data} = await post(
  'https://api.notion.com/v1/pages', {
  "parent": { "database_id": databaseID },
  "properties": {
    "Name": {
      "title": [
        {
          "text": {
            "content": content
          }
        }
      ]
    }
  }
},
{
  headers: {
    Authorization: `Bearer ${notionToken}`,
    "Content-Type": "application/json",
    "Notion-Version": "2021-05-13"
  }
})

let bodyContent = await arg('Enter Content')

let {bodyData} = await patch(
  `https://api.notion.com/v1/blocks/${data.id}/children`, {
  "children": [
    {
      "object": "block",
      "type": "paragraph",
      "paragraph": {
        "text": [
          {
            "type": "text",
            "text": {
              "content": bodyContent
            }
          }
        ]
      }
    }
  ]
},
{
  headers: {
    Authorization: `Bearer ${notionToken}`,
    "Content-Type": "application/json",
    "Notion-Version": "2021-05-13"
  }
})

let pageID = data.id.replace(/-/g, "");
let pageURL = `https://notion.so/${databaseID}?p=${pageID}`

copy(pageURL)
```

(I just noticed that I haven't updated the first script to pass the "Notion-Version" header, but it seems unnecessary for now.)

**Comment by johnlindquist** [comment]
Definitely going to check this out.

A multiline input makes a lot of sense. I'll try to get that in end of this week. 👍

**Reply by dealingwith** [reply]
Ask and ye shall receive!
